### ISSUE TITLE
Unable to build the ios app.

### ISSUE THREAD

**Issue by saileshbro** [DeepARSDK/deepar-flutter-plugin#78]
@samyakkkk @wilson-christian @mohamedsalah414 do you know what might be the issue here?

```
Launching lib/main_development.dart on iPhone 13 Pro Max in debug mode...
main_development.dart:1
Xcode build done.                                           13.0s
Failed to build iOS app
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_ARView

Error (Xcode): Undefined symbol: _OBJC_CLASS_$_CameraController

Error (Xcode): Undefined symbol: _OBJC_CLASS_$_DeepAR

Could not build the application for the simulator.
Error launching application on iPhone 13 Pro Max.
Exited
```

**Comment by saileshbro** [comment]
Not sure if this was necessary but i copied the `DeepAR.xcworkspace` to the `.pub-cache` dir as well for ios.

```bash
#!/bin/bash
function parse_yaml {
  local prefix=$2
  local s='[[:space:]]*' w='[a-zA-Z0-9_]*' fs=$(echo @ | tr @ '\034')
  sed -ne "s|^\($s\):|\1|" \
    -e "s|^\($s\)\($w\)$s:$s[\"']\(.*\)[\"']$s\$|\1$fs\2$fs\3|p" \
    -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p" $1 |
    awk -F$fs '{
      indent = length($1)/2;
      vname[indent] = $2;
      for (i in vname) {if (i > indent) {delete vname[i]}}
      if (length($3) > 0) {
         vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
         printf("%s%s%s=\"%s\"\n", "'$prefix'",vn, $2, $3);
      }
   }'
}
# Extract the version of the deepar_flutter package from pubspec.yaml
DEEPAR_VERSION=$(parse_yaml pubspec.lock | grep deepar_flutter_version | cut -d '"' -f 2)

ANDROID_TARGET="$HOME/.pub-cache/hosted/pub.dartlang.org/deepar_flutter-$DEEPAR_VERSION/android/libs/"
IOS_TARGET="$HOME/.pub-cache/hosted/pub.dartlang.org/deepar_flutter-$DEEPAR_VERSION/ios/libs/"

# Set the paths to the Android file and target directory
ANDROID_FILE="android/app/libs/deepar.aar"
IOS_FILE="ios/DeepAR.xcframework"

# Create the target directory if it doesn't exist
mkdir -p $ANDROID_TARGET
mkdir -p $IOS_TARGET

# Copy the Android file to the target directory
rsync $ANDROID_FILE $ANDROID_TARGET
rsync -r $IOS_FILE $IOS_TARGET
```

Also updated the `Podfile` for platform to be `13.0+`
```ruby
platform :ios, '14.0'
```

For anyone who is trying to automate this process.

**Comment by saileshbro** [comment]
Still, only works in the device, not in the simulator.

**Comment by saileshbro** [comment]
Apparently, this https://github.com/DeepARSDK/deepar-flutter-plugin/issues/78#issue-1592955798 happens only in the simulator. The app is building fine on a real device.

**Comment by saileshbro** [comment]
@mohamedsalah414 @samyakkkk @wilson-christian Any update on this one?

**Comment by wilson-christian** [comment]
Hi @saileshbro please try following : 

1. Download latest DeepAR SDK and copy `simulator>DeepAR.framework` folder and paste it at `pub-cache>deepar-flutter-0.0.5>ios` this will replace the older framework.
2. Inside your iOS project on flutter side open `Podfile` and add `config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64'` 
3. Clean project and install pods again.

Let me know if this works for you.

**Comment by wilson-christian** [comment]
<img width="791" alt="Screenshot 2023-02-28 at 4 31 04 PM" src="https://user-images.githubusercontent.com/59436377/221835049-a4c8a499-c19f-4913-ad35-3be0efaa1ad3.png">

**Comment by saileshbro** [comment]
Hey @wilson-christian Still getting the same error while building.
![image](https://user-images.githubusercontent.com/43571990/221839799-4857721a-62ce-4fc3-b536-6d74dd9ec363.png)
Here's my `.pub-cache`
<img width="739" alt="image" src="https://user-images.githubusercontent.com/43571990/221840824-e7d3dda2-9d9b-47ae-b7f5-a9e66aa22568.png">

And, here's my `Podfile`
<img width="940" alt="Screenshot 2023-02-28 at 5 13 34 PM" src="https://user-images.githubusercontent.com/43571990/221840982-a19d5022-7acf-4f0b-b40d-648a1ac8c913.png">



Let me know if I am missing anything. I am unable to get it running in the simulator.

**Comment by wilson-christian** [comment]
@saileshbro can you confirm this is how your pub-cache `deepar-flutter` folder looks?

<img width="358" alt="Screenshot 2023-02-28 at 7 35 09 PM" src="https://user-images.githubusercontent.com/59436377/221877336-a4b94e4a-a5c7-4be7-a576-a1052717c61c.png">

**Comment by saileshbro** [comment]
@wilson-christian Its different actually, I just ran `flutter pub cache clean` and then `flutter pub get`
It downloaded the `deepar_flutter-0.0.5` inside `.pub-cache/hosted/pub.dev` instead of `.pub-cache/hosted/pub.dartlang.org`.

I continued the setup for `pub.dev` its not running.

I tried to move the `deepar_flutter-0.0.5` (having the downloaded DeepAR SDK files) from `pub.dev` to `pub.dartlang.org`, now its showing dependency issues, need to run `flutter pub get`.

When i run `flutter pub get` now, its loading the `deepar_flutter-0.0.5` inside `pub.dev`.

**Comment by saileshbro** [comment]
Even if i have the same config in both folders, `pub.dev` and `pub.dartlang.org`, i get the same error.
Here are the contents of the file.
<img width="941" alt="image" src="https://user-images.githubusercontent.com/43571990/221892857-0fe345ee-b609-470f-a0dc-c675cec75cb5.png">

<img width="942" alt="image" src="https://user-images.githubusercontent.com/43571990/221892969-7160a7d9-2a81-4c9c-a254-7ad62d9b0e3b.png">

> NOTE: Here i have manually created `pub.dartlang.org` folder, because `flutter pub get` didn't create one for me.

**Comment by wilson-christian** [comment]
@saileshbro before moving further, please confirm that you have copied the framework from `SIMULATOR` folder from the downloaded SDK. Can you confirm please?

**Comment by saileshbro** [comment]
@wilson-christian yes confirmed, still the same error, I unzipped again, removed the `deepar_flutter` from `pub.dev`, ran `flutter pub get` again,
moved the `DeepAr.framework` from `simulator` to `deepar_flutter-0.0.5/ios/libs/DeepAR.framework`.
Still getting the same error
```
Launching lib/main_development.dart on iPhone 13 Pro Max in debug mode...
Running pod install...                                              9.5s
Running Xcode build...                                          
Xcode build done.                                           16.9s
Failed to build iOS app
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_ARView

Error (Xcode): Undefined symbol: _OBJC_CLASS_$_CameraController

Error (Xcode): Undefined symbol: _OBJC_CLASS_$_DeepAR

Could not build the application for the simulator.
Error launching application on iPhone 13 Pro Max.
```

**Comment by saileshbro** [comment]
@wilson-christian Can we bundle the SDK inside the pub package itself, this will fix all the issues may be?

**Comment by wilson-christian** [comment]
@saileshbro in latest `0.0.5` we have already bundle the framework with pub, but due to size limitation on `pub` we are unable to attach framework for iOS. For real device everything will work fine.

**Comment by wilson-christian** [comment]
I noticed you are pasting `DeepAR.framework` in `ios/lib` but instead it should be just inside `ios` folder. Check my folder structure image I sent above.

**Comment by saileshbro** [comment]
@wilson-christian this isn't very clear. `DeepAR.framework` was already present when I ran `flutter pub get`. Does this mean I don't need to do anything extra for the iOS setup as I did for android (copying the `.aar` file in `android/libs/`?

Still getting the same issue, tho.

**Comment by saileshbro** [comment]
> @saileshbro can you confirm this is how your pub-cache `deepar-flutter` folder looks?
> 
> <img alt="Screenshot 2023-02-28 at 7 35 09 PM" width="358" src="https://user-images.githubusercontent.com/59436377/221877336-a4b94e4a-a5c7-4be7-a576-a1052717c61c.png">

![image](https://user-images.githubusercontent.com/43571990/221903954-e43c2eb4-0c31-4224-8e7b-66feb9c8909c.png)

This is the fresh `flutter pub get`

**Comment by saileshbro** [comment]
Finally got it working, had to replace the `~/.pub-cache/hosted/pub.dev/deepar_flutter-0.0.5/ios/DeepAR.framework` with the `DeepAR-iOS-v4.1.0/lib/simulator/DeepAR.framework`.

> NOTE: Please make sure where your `deepar_flutter-0.0.5` is located, the documentation says it's inside `pub.dartlang.org` but for me it was inside `pub.dev`.

**Comment by wilson-christian** [comment]
@saileshbro this process will enable you to run app on simulator. But please note, 

**VERY IMPORTANT :  When you want to release the app, clean pub-cache and download the pub again to get the default `DeepAR.framework` back as it is required to create production apps.**
